src/vxfw/Border.zig

```diff
@@ -65,6 +75,33 @@ pub fn draw(self: *const Border, ctx: vxfw.DrawContext) Allocator.Error!vxfw.Sur
         surf.writeCell(0, row, .{ .char = .{ .grapheme = "│", .width = 1 }, .style = self.style });
         surf.writeCell(right_edge, row, .{ .char = .{ .grapheme = "│", .width = 1 }, .style = self.style });
     }
+
+    for (self.labels) |label| {
+        const text_len: u16 = @intCast(label.text.len);
+        const text_row: u16 = switch (label.alignment) {
+            .TopLeft, .TopCenter, .TopRight => 0,
+            .BottomLeft, .BottomCenter, .BottomRight => bottom_edge,
+        };
+
+        const text_col: u16 = switch (label.alignment) {
+            .TopLeft, .BottomLeft => 1,
+            .TopCenter, .BottomCenter => @max((size.width - text_len) / 2, 1),
+            .TopRight, .BottomRight => @max(size.width - 1 - text_len, 1),
+        };
+
+        var i: usize = 0;
+        while (i < label.text.len and text_col + i < size.width - 1) : (i += 1) {
```

Comment:
This loop needs to work on graphemes, not just bytes. The `DrawContext` object available in this function has a method called `graphemeIterator` which you can use to iterate over the label's graphemes:

```zig
            var iter = ctx.graphemeIterator(label.text);
            while (iter.next()) |grapheme| {
                ...
            }
```